### PR TITLE
[WIP] Re-enable DPI scaling on GTK

### PR DESCRIPTION
**Pull request by MartinZikmund**
GitHub Issue (If applicable): closes #

## PR Type

What kind of change does this PR introduce?

- Bugfix

## What is the current behavior?

**TODO Does not support fractional scaling correctly yet.**

## PR Checklist

Please check if your PR fulfills the following requirements:

- [ ] Docs have been added/updated which fit [documentation template](https://github.com/unoplatform/uno/blob/master/doc/.feature-template.md) (for bug fixes / features)
- [ ] [Unit Tests and/or UI Tests](https://github.com/unoplatform/uno/blob/master/doc/articles/uno-development/working-with-the-samples-apps.md) for the changes have been added (for bug fixes / features) (if applicable)
- [ ] Validated PR `Screenshots Compare Test Run` results.
- [ ] Contains **NO** breaking changes
- [ ] Associated with an issue (GitHub or internal) and uses the [automatic close keywords](https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue).
- [ ] Commits must be following the [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/#summary) specification.

## Other information

Internal Issue (If applicable):

### CODE REVIEW

File: src/Uno.UI.Runtime.Skia.Gtk/GLRenderSurface.cs [hunk 1]
```diff
@@ -83,10 +83,12 @@ private void UnoGLDrawingArea_Render(object o, RenderArgs args)
 				_grContext = GRContext.CreateGl(glInterface);
 			}
 
+			var dpi = _dpi ?? 1f;
+			var scaledGuardBand = (int)(GuardBand * dpi);
 			// manage the drawing surface
 			var res = (int)Math.Max(1.0, Screen.Resolution / 96.0);
```
Comment (spouliot) [inline src/Uno.UI.Runtime.Skia.Gtk/GLRenderSurface.cs:89]: is `res` still used anywhere ?

File: src/Uno.UI.Runtime.Skia.Gtk/GLRenderSurface.cs [hunk 2]
```diff
@@ -125,9 +127,16 @@ private void UnoGLDrawingArea_Render(object o, RenderArgs args)
...
+
+				if (_dpi != null)
+				{
+					canvas.SetMatrix(SKMatrix.CreateScale((float)dpi, (float)dpi));
+				}
 				canvas.Translate(new SKPoint(0, GuardBand));
 
 				WUX.Window.Current.Compositor.Render(_surface);
+
+				canvas.SetMatrix(SKMatrix.CreateScale((float)1 / dpi, (float)1 / dpi));
```
Comment (spouliot) [inline src/Uno.UI.Runtime.Skia.Gtk/GLRenderSurface.cs:139]: I do not think this is required. The call to `SKAutoCanvasRestore.Dispose()` should reset the matrix of the canvas.

File: src/Uno.UI.Runtime.Skia.Gtk/GLRenderSurface.cs [hunk 3]
```diff
@@ -83,10 +83,12 @@ private void UnoGLDrawingArea_Render(object o, RenderArgs args)
 				_grContext = GRContext.CreateGl(glInterface);
 			}
 
+			var dpi = _dpi ?? 1f;
+			var scaledGuardBand = (int)(GuardBand * dpi);
```
Comment (spouliot) [inline src/Uno.UI.Runtime.Skia.Gtk/GLRenderSurface.cs:87]: this could be updated inside `UpdateDpi` (which should not happen frequently)

**Review by spouliot — COMMENTED** [review]
Nice, I had noticed that `_dpi` was not used in this renderer :-)

File: src/Uno.UI.Runtime.Skia.Gtk/GLRenderSurface.cs (hunk 1 above)
Comment (jeromelaban) [reply src/Uno.UI.Runtime.Skia.Gtk/GLRenderSurface.cs]: Fixed in recent changed

File: src/Uno.UI.Runtime.Skia.Gtk/GLRenderSurface.cs (hunk 2 above)
Comment (jeromelaban) [reply src/Uno.UI.Runtime.Skia.Gtk/GLRenderSurface.cs]: Fixed in recent changes

**Review by jeromelaban — APPROVED** [review]

File: src/Uno.UI.Runtime.Skia.Gtk/GLRenderSurface.cs (hunk 3 above)
Comment (jeromelaban) [reply src/Uno.UI.Runtime.Skia.Gtk/GLRenderSurface.cs]: Will fix in subsequent update

**Pull request merged**